src/Symfony/Component/String/UnicodeString.php

```diff
@@ -84,6 +84,16 @@ public function chunk(int $length = 1): array
         return $chunks;
     }
 
+    /**
+     * @return int[]|null
+     */
+    public function codePoints(int $offset = 0): ?array
```

Comment:
What about moving this to AbstractUnicodeString? I'd rename it `codePointsAt` and make the argument mandatory: that might make it easier to discover what this is about.
Also, this should return the empty array instead of null when the offset is out of bound.

Reply:
> For me it does not make sense to use this method in a CodePointString, since every offset is representing a single code point.

It does because it's at the abstract level: abstract means it should generic enough to cover the domain, which is what this would provide. Putting the method at the abstract level is way better than forcing one to rely on more specialized methods.

>  But I think then we should also rename codePoint to codePointAt for consistency.

I would remove codePointAt entirely

> I don't agree. Imo it is better to distinguish 'the array of codepoints for the grapheme at requested offset' from 'the offset does not exist'.

sorry but I insist, having a nullable return type doesn't make sense, the more specific the better.

Reply:
> It does because it's at the abstract level: abstract means it should generic enough to cover the domain, which is what this would provide. Putting the method at the abstract level is way better than forcing one to rely on more specialized methods.

> I would remove codePointAt entirely

ok

> sorry but I insist, having a nullable return type doesn't make sense, the more specific the better.

Ok, will change the return type (while still disagreeing, but that's ok ;) )

Reply:
should we actually return null or empty array for an out of bound index, or throw an `\OutOfBoundsException` ?

Reply:
I think `\OutOfBoundsException` would be better indeed. What do you think @nicolas-grekas ?

Reply:
I tried to avoid throwing unnecessary exception as much as possible so throwing wouldn't be consistent with the rest of the component to me.

Reply:
Ok, I will work on it